$NetBSD: patch-configure,v 1.4 2024/06/18 09:43:23 markd Exp $

(1)
Convert install path from fvwm to fvwm2
This can not be done with
CONFIGURE_ARGS+=        --datarootdir=${PREFIX:Q}/share/fvwm2
this will install fvwm2/fvwm.

(2)
Fixes for C99 compatibility from Fedora
A cross build cannot run the keysym and libstroke tests: assume current X11 and libstroke. IRIX declares an m_sync().

--- configure.orig
+++ configure
@@ -609,7 +609,7 @@
 MAKEFLAGS=
 
 # Identity of this package.
-PACKAGE_NAME='fvwm'
+PACKAGE_NAME='fvwm2'
 PACKAGE_TARNAME='fvwm'
 PACKAGE_VERSION='2.7.0'
 PACKAGE_STRING='fvwm 2.7.0'
@@ -3698,7 +3698,7 @@
 
 
 # Define the identity of the package.
- PACKAGE='fvwm'
+ PACKAGE='fvwm2'
  VERSION='2.7.0'
 
 
@@ -7643,9 +7643,8 @@
 int
 main (void)
 {
-int i; static j; int *p; char *c;
-  switch (*p = p = *c) { case 0: printf("%Q", c, p); }
-  *c = &i; c = p;
+int unused; int *p; char *c;
+  printf("%Q", c, p);
   while (1 || (unsigned int)3 >= 0 || ((int)-1) == ((unsigned int)1));
 
   ;
@@ -13512,7 +13511,7 @@
 int
 main (void)
 {
-extern select ($ac_cv_type_fd_set_size_t,
+extern int select ($ac_cv_type_fd_set_size_t,
  $ac_cv_type_fd_set *,	$ac_cv_type_fd_set *, $ac_cv_type_fd_set *,
  $ac_type_timeval *);
   ;
@@ -13615,6 +13614,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
 int main(void)
 {
   char template[128];
@@ -13940,6 +13942,9 @@
 
 if test "$cross_compiling" = yes
 then :
+  : # cross: assume X11 has XK_Page_Up (no COMPAT_OLD_KEYSYMDEF)
+elif false
+then :
   { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -13983,6 +13988,9 @@
 
 if test "$cross_compiling" = yes
 then :
+  : # cross: assume a current libstroke (no COMPAT_OLD_LIBSTROKE)
+elif false
+then :
   { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 printf "%s\n" "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
